robot-server/robot_server/runs/run_controller.py

```diff
@@ -0,0 +1,95 @@
+"""Control an active run with Actions."""
+import logging
+from datetime import datetime
+
+from opentrons.protocol_engine import ProtocolEngineError
+
+from robot_server.service.task_runner import TaskRunner
+
+from .engine_store import EngineStore
+from .run_store import RunStore
+from .action_models import RunAction, RunActionType
+
+
+log = logging.getLogger(__name__)
+
+
+class RunActionNotAllowedError(ValueError):
+    """Error raised when a given run action is not allowed."""
+
+
+class RunController:
+    """An interface to manage the side-effects of requested run actions."""
+
+    def __init__(
+        self,
+        run_id: str,
+        task_runner: TaskRunner,
+        engine_store: EngineStore,
+        run_store: RunStore,
+    ) -> None:
+        self._run_id = run_id
+        self._task_runner = task_runner
+        self._engine_store = engine_store
+        self._run_store = run_store
+
+    def create_action(
+        self,
+        action_id: str,
+        action_type: RunActionType,
+        created_at: datetime,
+    ) -> RunAction:
+        """Create a run action.
+
+        Arg:
+            run_id: The run associated to the action.
+            create_action: Action to create.
+
+        Returns:
+            The action that was created.
+
+        Raises:
+            RunNotFoundError: The given run identifier was not found in the database.
+            RunStopped: The given run is not the current run.
+            RunActionNotAllowed: The following operation is not allowed
```

Comment:
Correct assertions are raised by method, but docstring is out of date.

```suggestion
            RunActionNotAllowed: The requested action type is not currently allowed
```